src/lib/replication/repl_dev/raft_repl_dev.h

```diff
@@ -86,6 +89,8 @@ class RaftReplDev : public ReplDev,
 
     RaftReplDevMetrics m_metrics;
 
+    nuraft::ptr< nuraft::snapshot > m_last_snapshot{nullptr};
```

Comment:
Why do homestore need to maintain the last snapshot. Isn't upper layer going to maintain snapshot. It could potentially store multiple snapshots?

Reply:
You point is valid. I also thought about it, I did this only because raft is asking for a nuraft::snapshot and we don't want to expose to upper layer, and in this snapshot, there is only log_idx and log_term which is not mutable to upper layer (right?), so I didn't ask for upper layer to get the last snapshot.